common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -124,6 +137,172 @@ def test_add_entitlement(self):
+        )
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+        self._assert_default_policy(course_entitlement.policy)
+
+    def test_default_no_matching_policy_entry(self):
+        """
+        Verify that, when no course entitlement policy is found with the same mode or site
+        as the created entitlement, the default policy is used for the entitlement.
+        """
+        CourseEntitlementPolicy.objects.create(mode=CourseMode.PROFESSIONAL, site=None)
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+        self._assert_default_policy(course_entitlement.policy)
+
+    def test_set_custom_mode_policy_on_create(self):
+        """
+        Verify that, when there does not exist a course entitlement policy with the same mode and site as
+        a created entitlement, but there does exist a policy with the same mode and a null site,
+        that policy is assigned to the entitlement.
+        """
+        policy = CourseEntitlementPolicy.objects.create(mode=CourseMode.PROFESSIONAL, site=None)
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+        entitlement_data['mode'] = CourseMode.PROFESSIONAL
+
+        self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+        assert course_entitlement.policy == policy
+
+    def test_set_custom_site_policy_on_create(self):
+        """
+        Verify that, when there does not exist a course entitlement policy with the same mode and site as
+        a created entitlement, but there does exist a policy with the same site and a null mode,
+        that policy is assigned to the entitlement.
+        """
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+
+        policy_site = SiteFactory.create()
+        policy = CourseEntitlementPolicy.objects.create(mode=None, site=policy_site)
+
+        set_entitlement_policy(course_entitlement, policy_site)
```

Comment:
Is the goal of this test to test that the policy is applied during the POST or through the use of the set_entitlement_policy method?

It seems to use/partially test both.

Reply:
The reason I use the POST call is just for simplicity sake of creating the entitlement product with data from the existing `_get_data_set` helper method.
Unfortunately, because of not being able to pull out the request's site or manually assign it in the unit test I had to separately test just the `set_entitlement_policy` helper method to confirm functionality incorporating site-specificity into policy selection.

Reply:
Ah ok I remember this conversation now.  We might want to add a comment here to highlight this reasoning.